### PR TITLE
CRYPTO-3: Change default cipher as OpensslCipher

### PR DESCRIPTION
**Pull request by winningsix**
This fixes [Change default cipher as OpensslCipher](https://issues.apache.org/jira/browse/CRYPTO-3).

### CODE REVIEW

File: src/main/java/com/intel/chimera/cipher/CipherFactory.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public static Cipher getInstance(CipherTransformation transformation,
       }
     }
 
-    return (cipher == null) ? new JceCipher(props, transformation) : cipher;
+    return (cipher == null) ? new OpensslCipher(props, transformation) : cipher;
```
Comment (jerrychenhf) [inline src/main/java/com/intel/chimera/cipher/CipherFactory.java]: I would consider this change is dangerous.
Change default cipher to OpensslCipher makes sense but change to OpensslCipher when no Cipher is working may be dangerous. I would suggest, we still use JceCipher here.

File: src/main/java/com/intel/chimera/cipher/CipherFactory.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ public static Cipher getInstance(CipherTransformation transformation,
    */
   public static Cipher getInstance(CipherTransformation transformation)
       throws GeneralSecurityException {
-    return getInstance(transformation, null);
+    return getInstance(transformation, new Properties());
```
Comment (jerrychenhf) [inline src/main/java/com/intel/chimera/cipher/CipherFactory.java]: Is this change needed?

File: src/main/java/com/intel/chimera/utils/Utils.java [hunk 3]
```diff
@@ -135,8 +135,9 @@ public static int getBufferSize(Properties props) {
    */
   public static String getCipherClassString(Properties props) {
     final String configName = CHIMERA_CRYPTO_CIPHER_CLASSES_KEY;
-    return props.getProperty(configName) != null ? props.getProperty(configName) : System
-        .getProperty(configName, CHIMERA_CRYPTO_CIPHER_CLASSES_DEFAULT);
+    return props.getProperty(configName) != null ? props
```
Comment (jerrychenhf) [inline src/main/java/com/intel/chimera/utils/Utils.java]: If we do want to handle the case that if am empty class string is specified, we want to default to OpensslCipher,
We can check empty string here and set to CLASSES_DEFAULT for empty value.

**Comment by winningsix** [comment]
Thanks  @jerrychenhf  for the reviews. Patch is updated addressing your comments.

File: src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java [hunk 4]
```diff
@@ -47,6 +50,9 @@
    */
   public static Cipher getInstance(CipherTransformation transformation,
       Properties props) throws GeneralSecurityException {
+    if(props == null){
```
Comment (jerrychenhf) [inline src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java]: I would suggest to put the check of props == null into Utils.getCipherClasses method. 
This can centralize the point of handling null props and avoid the need of creating a new Properties object.

File: src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java [hunk 5]
```diff
@@ -80,14 +86,15 @@ public static Cipher getInstance(CipherTransformation transformation)
...
 
+  // Return OpenSSLCipher if Properties is null or empty by default
   private static List<Class<? extends Cipher>> getCipherClasses(Properties props) {
     List<Class<? extends Cipher>> result = new ArrayList<Class<? extends
         Cipher>>();
     String cipherClassString = Utils.getCipherClassString(props);
-    if (cipherClassString == null) {
-      LOG.debug("No cipher classes configured.");
-      return null;
+    if (cipherClassString.isEmpty()) {
```
Comment (jerrychenhf) [inline src/main/java/org/apache/commons/crypto/cipher/CipherFactory.java]: Suggest to put this to centralized place: Utils.getCipherClassString.

**Comment by winningsix** [comment]
Hi @jerrychenhf, I have updated this PR. Please help me review it. Thank you!

**Comment by jerrychenhf** [comment]
+1

**Comment by winningsix** [comment]
Thanks @jerrychenhf for your review. Committed to master at c0df815
